### DISCUSSION TITLE
Which widgets allow cursor highlighting and copying?

### DISCUSSION THREAD

**Discussion by showipintbri** [Textualize/textual#1843] [Q&A] [answered]
I've made a small app that generates data (flat text) in a `Static()` widget.

I want to be able to use the mouse to highlight text and right-click for copy, or use CTRL+C to copy.

Is highlighting and/or copying text from a Static() widget available?
Is there a way to achieve the outcome using any available features, functions, or other widgets?

**Comment by davep** [comment, marked as answer]
Generally copying text from the application is something that is down to your terminal, and how that works can differ between terminals. Textual itself doesn't have a method of placing text in your operating system's clipboard.

I've not tested it myself but I imagine it would be possible to add something like that to your application with a library such as [pyperclip](https://pypi.org/project/pyperclip/).

**Reply by showipintbri** [reply]
Thanks, I wasn't sure if I'd need an external library or not. I've used pyperclip before so I'm sure I can make something work move data between the app and clipboard.

I'm still struggling to get anything to highlight on the screen. Is anyone else experiencing this or have a solution?

Running Textual and Python in a WSL2, accessing it via Windows Command Prompt (albeit renders a terrible UI), also the new "Windows Terminal" (renders a beautiful UI). Neither seem to allow highlighting while in a Textual app, however all other highlighting works when working within in the WSL2 UbuntuOS.

**Reply by davep** [reply]
> I'm still struggling to get anything to highlight on the screen. Is anyone else experiencing this or have a solution?

This is quite normal for terminal applications that place the terminal in "application mode" -- keep in mind that you can use the mouse to interact with the application. Most terminals have some sort of system that allows for making and copying. On macOS I use iTerm and the default configuration there is to hold down the <kbd>Shift</kbd> key while marking text.

**Reply by showipintbri** [reply]
@davep Thanks for your excellent explanations and expertise. I'll continue to investigate with different terminals and systems.

**Reply by davep** [reply]
Most welcome. All being well we should end up with a more comprehensive FAQ entry at some point: https://github.com/Textualize/textual/issues/1848
